definitions/mw.lua

```diff
@@ -0,0 +1,649 @@
+---@return boolean
+function mw.isSubsting() end
+
+---See www.mediawiki.org/wiki/Extension:Scribunto/Lua_reference_manual#mw.loadData
+---@param module string
+---@return table
+function mw.loadData(module) end
+
+---This is the same as mw.loadData(), except it loads data from JSON pages rather than Lua tables. The JSON content must be an array or object. See also mw.text.jsonDecode().
+---@param page string
+---@return table
+function mw.loadData(page) end
+
+---Serializes object to a human-readable representation, then returns the resulting string.
+---@param object any
+---@return string
+function mw.dumpObject(object) end
+
+---Passes the arguments to mw.allToString(), then appends the resulting string to the log buffer.
+---@param ... any
+function mw.log(...) end
+
+---Calls mw.dumpObject() and appends the resulting string to the log buffer. If prefix is given, it will be added to the log buffer followedby an equals sign before the serialized string is appended (i.e. the logged text will be "prefix = object-string").
+---@param object any
+---@param prefix any?
+function mw.logObject(object, prefix) end
+
+---@class Frame
+---@field args table
+mw.frame = {}
+
+---Call a parser function, returning an appropriate string. This is preferable to frame:preprocess, but whenever possible, native Lua functions or Scribunto library functions should be preferred to this interface.
+---@param name string
+---@param args table|string
+---@return string
+---@overload fun(self, params: {name: string, args: table|string}): string
+---@overload fun(self, name: string, ...: string): string
+function mw.frame:callParserFunction(name, args) end
+
+---This is transclusion. As in transclusion, if the passed title does not contain a namespace prefix it will be assumed to be in the Template: namespace.
+---@param params {title: string, args: table}
+---@return string
+function mw.frame:expandTemplate(params) end
+
+---This is equivalent to a call to frame:callParserFunction() with function name '#tag:' .. name and with content prepended to args.
+---@param name string
+---@param content string
+---@param args table|string
+---@return string
+---@overload fun(self, params: {name: string, content: string, args: table|string}): string
+function mw.frame:extensionTag(name, content, args) end
+
+---Called on the frame created by {{#invoke:}}, returns the frame for the page that called {{#invoke:}}. Called on that frame, returns nil.
+---@return Frame
+function mw.frame:getParent() end
+
+---Returns the title associated with the frame as a string. For the frame created by {{#invoke:}}, this is the title of the module invoked.
+---@return string
+function mw.frame:getTitle() end
+
+---Returns the title associated with the frame as a string. For the frame created by {{#invoke:}}, this is the title of the module invoked.
+---@param params {title: string, args: table}
+---@return string
+function mw.frame:newChild(params) end
```

Comment:
`Returns the title associated with the frame as a string. For the frame created by {{#invoke:}}, this is the title of the module invoked.`
belongs to the function above
```suggestion
---Create a new Frame object that is a child of the current frame, with optional arguments and title.
--This is mainly intended for use in the debug console for testing functions that would normally be called by {{#invoke:}}. The number of frames that may be created at any one time is limited.
---@param params {title: string, args: table}
---@return string
function mw.frame:newChild(params) end
```